incubator/hnc/internal/validators/object.go

```diff
@@ -37,36 +38,52 @@ type Object struct {
 }
 
 func (o *Object) Handle(ctx context.Context, req admission.Request) admission.Response {
-	log := o.Log.WithValues("nm", req.Name, "nnm", req.Namespace)
+	log := o.Log.WithValues("nm", req.Name, "resource", req.Resource, "nnm", req.Namespace, "op", req.Operation)
+
+	// Before even looking at the objects, early-exit for any changes we shouldn't be involved in.
+	// This reduces the chance we'll hose some aspect of the cluster we weren't supposed to touch.
+	//
+	// Firstly, skip namespaces we're excluded from (like kube-system).
+	if config.EX[req.Namespace] {
+		return allow("excluded namespace " + req.Namespace)
+	}
 	// Allow changes to the types that are not in propagate mode. This is to dynamically enable/disable
 	// object webhooks based on the types configured in hncconfig. Since the current admission rules only
 	// apply to propagated objects, we can disable object webhooks on all other non-propagate-mode types.
 	if !o.isPropagateType(req.Kind) {
 		return allow("Non-propagate-mode types")
 	}
+	// Finally, let the HNC SA do whatever it wants.
 	if isHNCServiceAccount(&req.AdmissionRequest.UserInfo) {
 		log.V(1).Info("Allowed change by HNC SA")
 		return allow("HNC SA")
 	}
 
+	// Decode the old and new object. We validate both creations and deletions, so either of these can
```

Comment:
'We validate both creations and deletions' seems confusing here. Maybe 'Either of these can be missing for creations or deletions'?

Reply:
Done